src/seedu/addressbook/data/person/Address.java

```diff
@@ -13,20 +13,50 @@
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+
+	public final Block block;
+    public final Street street;
+    public final Unit unit;
+    public final PostalCode postalCode;
     private boolean isPrivate;
 
     /**
      * Validates given address.
-     *
+     * Minimal address needs a Block and Street
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+    	String[] parts = address.split(",");
+    	
+    	this.block = new Block(Integer.parseInt(parts[0]));
+    	this.street = new Street(parts[1]);
+    	this.unit = (parts.length > 2) ? new Unit(parts[2]) : new Unit("");
+    	this.postalCode = (parts.length > 3) ? new PostalCode(parts[3]) : new PostalCode("");
+    	
+        this.value = this.block.value + "," +  this.street.value + 
+        		((this.unit.value == "") ? "" : "," + this.unit.value) + 
+        		((this.postalCode.value == "") ? "" : "," + this.postalCode.value);
+        
```

Comment:
Keeping this variable (value) seems redundant and may cause future problems like inconsistent data as you have introduced new ways of handling the address value.